src/source/src_hops.c

```diff
@@ -133,6 +157,57 @@
 
     }
 
+    void src_hops_open_interface_pulseaudio(src_hops_obj * obj) {
+
+        pa_sample_format_t format;
+
+        switch (obj->format->type) {
+            
+            case format_binary_int08:
+
+                format = PA_SAMPLE_U8;
+
+            break;
+
+            case format_binary_int16:
+
+                format = PA_SAMPLE_S16LE;
+
+            break;
+
+            case format_binary_int24:
+
+                format = PA_SAMPLE_S24LE;
+
+            break;
+            
+            case format_binary_int32:
+            
+                format = PA_SAMPLE_S32LE;
+            
+            break;
+
+            default:
+
+                printf("Source hops: Invalid format.\n");
+                exit(EXIT_FAILURE);
```

Comment:
That would be nice to create a MACRO for this pattern which is everywhere. 
I would not do it now, since every file would be modified.

Reply:
Minimum effort for now is ok!

Reply:
Ok, won't change this